designs/0034-jacobian.md

```diff
@@ -0,0 +1,195 @@
+- Feature Name: `jacobian` target
+- Start Date: 01-25-2024
+
+# Summary
+[summary]: #summary
+
+This design doc proposes adding a `jacobian` target and `*_jacobian` user defined functions. The `jacobian` target will be accessible directly in `transformed parameters` and the new `*_jacobian` functions.
+
+
+The examples below show an example use case.
+
+```stan
+functions {
+  real upper_bound_jacobian(real x, real upper_bound) {
+    jacobian += x;
+    return upper_bound - exp(x);
+  }
+}
+data {
+ real ub;
+}
+parameters {
+ // User defined constraint
+ real b_raw;
+}
+transformed paramters {
+  real b = upper_bound_jacobian(b_raw, ub);
+}
+```
+
+# Motivation
+[motivation]: #motivation
+
+Given a function $c$ mapping unconstrained parameters in $Y$ to constrained parameters in $X$, probability density function $\pi$, and a Jacobian determinant function over the constrained space $J(c(y))$, Stan calculates the log transformed density function [1]
+
+$$
+\pi^*(y) = \pi\left( c\left(y\right) \right) J_c\left(y\right)
```

Comment:
This needs to be | J(y)| -- it's an absolute determinant of the Jacobian evaluated at y.  Also in the next formula and in 44.  The _c isn't used in the explanation and that's fine.